src/envelope/spacecharge/EnvelopeSpaceChargePush.H

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_ENVELOPESPACECHARGEPUSH_H
+#define IMPACTX_ENVELOPESPACECHARGEPUSH_H
+
+#include "particles/ImpactXParticleContainer.H"
+#include "particles/CovarianceMatrix.H"
+
+#include <AMReX_REAL.H>
+
+
+namespace impactx::spacecharge
```

Comment:
You could move the `envelope_` prefix to a namespace here
```suggestion
namespace impactx::envelope::spacecharge
```
and remove the prefix from the function name below.

That way we mirror the directory structure, too, which is good.

(Same change then needs to go in the `.cpp` file and the locations we call this function from.

Reply:
updated.